cmd/earthly/main.go

```diff
@@ -1114,7 +1103,11 @@ func (app *earthlyApp) getAddressesForTCP(context *cli.Context) (addresses, erro
 		if app.cfg.Global.DebuggerHost != "" {
 			app.debuggerHost = app.cfg.Global.DebuggerHost
 		} else {
-			app.debuggerHost = fmt.Sprintf("tcp://%s:%v", bkURL.Hostname(), config.DefaultDebuggerPort)
+			if app.cfg.Global.DebuggerPort == config.DefaultDebuggerPort && bkURL.Scheme == "tcp" {
+				app.debuggerHost = fmt.Sprintf("tcp://%s:%v", bkURL.Hostname(), config.DefaultDebuggerPort)
+			} else {
+				app.debuggerHost = fmt.Sprintf("tcp://127.0.0.1:%v", app.cfg.Global.DebuggerPort)
+			}
```

Comment:
Since the old codepath needed to honor the port, we now support both. The  new, non-deprecated setting takes precedence, and the CLI wins over all. Port appears to no longer be available on the CLI?